File: algorithms/ruby/search/binary_search.rb
def binary_search(arr, target)
  low = 0
  high = arr.length - 1

  while low <= high
    mid = (low + high) / 2
    if arr[mid] == target
      return mid
    elsif arr[mid] < target
      low = mid + 1
    else
      high = mid - 1
    end
  end

  return -1
end

if __FILE__ == $0
  arr = [2, 3, 4, 10, 40]
  target = 10
  result = binary_search(arr, target)

  if result != -1
    puts "Element is present at index #{result}"
  else
    puts "Element is not present in array"
  end
end

File: algorithms/php/search/BinarySearch.php
<?php

function binarySearch($arr, $x) {
    if (count($arr) === 0) return false;
    $low = 0;
    $high = count($arr) - 1;
     
    while ($low <= $high) {
         
        // compute middle index
        $mid = floor(($low + $high) / 2);
 
        // element found at mid
        if($arr[$mid] == $x) {
            return $mid;
        }
 
        if ($x < $arr[$mid]) {
            // search the left side of the array
            $high = $mid -1;
        }
        else {
            // search the right side of the array
            $low = $mid + 1;
        }
    }
     
    // If we reach here element is x not in array
    return -1;
}
 
$arr = array(2, 3, 4, 10, 40);
$value = 10;
$result = binarySearch($arr, $value);
if($result == -1)
    echo "Element is not present in array";
else
    echo "Element is present at index " . $result;

?>
